mindsdb/integrations/handlers/groq_handler/README.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Groq
+sidebarTitle: Groq
+---
+
+This documentation describes the integration of MindsDB with [Groq](https://groq.com/), a cloud service that simplifies the way developers interact with cutting-edge LLMs through its API.
+
+## Prerequisites
+
+Before proceeding, ensure the following prerequisites are met:
+
+1. Install MindsDB locally via [Docker](https://docs.mindsdb.com/setup/self-hosted/docker) or [Docker Desktop](https://docs.mindsdb.com/setup/self-hosted/docker-desktop).
+2. To use Groq within MindsDB, install the required dependencies following [this instruction](https://docs.mindsdb.com/setup/self-hosted/docker#install-dependencies).
+3. Obtain the Groq API key required to deploy and use Groq models within MindsDB. Get the API key from [here](https://console.groq.com/keys).
+
+## Setup
+
+Create an AI engine from the [Groq handler](https://github.com/mindsdb/mindsdb/tree/main/mindsdb/integrations/handlers/groq_handler).
+
+```sql
+CREATE ML_ENGINE groq_engine
+FROM groq
+USING
+      groq_api_key = 'api-key-value'
+```
+
+Create a model using `groq_engine` as an engine.
+
+```sql
+CREATE MODEL groq_model
+PREDICT answer
+USING
+      engine = 'groq_engine',   -- engine name as created via CREATE ML_ENGINE
+      model_name = 'model-name',              -- choose one of available models
+      prompt_template = 'prompt-to-the-model' -- prompt message to be completed by the model
+```
+
+## Usage
+
+The following usage examples utilize `groq_engine` to create a model with the `CREATE MODEL` statement.
+
```

Comment:
Speaking of the supported parameters, does Groq also support the generation of images and embeddings? The OpenAI integration supports an `image` and `embedding` mode. If Groq does not support this, we might need to take some precautions. The configuration settings for the OpenAI integration are maintained [here](https://github.com/mindsdb/mindsdb/blob/main/mindsdb/integrations/handlers/openai_handler/constants.py) and then set as instance variables in the handler. Can you please have a look and make the necessary changes?

Reply:
There are no embedding and image models in Groq.

Reply:
So I need to override these instance variables like 
- supported_modes 
- chat_completion_models

Reply:
OK, then I think setting `self.supported_modes` in your handler to `['default', 'conversational', 'conversational-full']` should do the trick to prevent users from using these modes. Please test this once you make the changes.

Reply:
I think you have already overridden `chat_completion_models`, so you should be fine there.

Reply:
I am thinking of doing like this

<details>
<summary>Groq Handler Config</summary>

```py
class GroqHandlerConfig(BaseSettings):
    """
    Configuration for Groq handler.

    Attributes
    ----------

    BASE_URL : str
        Base URL for the Groq API.
    """

    BASE_URL: str = "https://api.groq.com/openai/v1"
    DEFAULT_MODEL: str = "llama3-8b-8192"
    DEFAULT_MODE: str = "default"
    SUPPORTED_MODES: list[str] = [
        'default',
        'conversational',
        'conversational-full',
    ]
    CHAT_MODELS: tuple[str] = (
        'gemma2-9b-it',
        'gemma-7b-it',
        'llama3-groq-70b-8192-tool-use-preview',
        'llama3-groq-8b-8192-tool-use-preview',
        'llama-3.1-70b-versatile',
        'llama-3.1-8b-instant',
        'llama-3.2-1b-preview',
        'llama-3.2-3b-preview',
        'llama-3.2-11b-vision-preview',
        'llama-3.2-90b-vision-preview',
        'llama-guard-3-8b',
        'llama3-70b-8192',
        'llama3-8b-8192',
        'mixtral-8x7b-32768'
    )
```

</details>

<details>
<summary>Groq Handler `__init__` function</summary>

```py
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.api_base = groq_handler_config.BASE_URL
        self.default_model = groq_handler_config.DEFAULT_MODEL
        self.default_mode = groq_handler_config.DEFAULT_MODE
        self.supported_modes = groq_handler_config.SUPPORTED_MODES
        self.chat_completion_models = groq_handler_config.CHAT_MODELS
```

</details>

Reply:
Hmm.. I think it is better to retrieve the chat_completion_models via an API call as you have done already. This is in case it changes in the future. Everything else looks good.